README.md
# Smoozzy HTTP Statuses

Collection of HTTP statuses as named constants

Status codes defined in:
- [RFC1945](https://tools.ietf.org/html/rfc1945) (HTTP/1.0)
- [RFC2616](https://tools.ietf.org/html/rfc2616) (HTTP/1.1)
- [RFC6585](https://tools.ietf.org/html/rfc6585) (Additional HTTP Status Codes)
- [RFC7538](https://tools.ietf.org/html/rfc7538) (Permanent Redirect)
- [RFC2324](https://tools.ietf.org/html/rfc2324) (Hyper Text Coffee Pot Control Protocol)
- [RFC2518](https://tools.ietf.org/html/rfc2518) (WebDAV)


## Usage

```javascript
import Api from '@smoozzy/api';
import {
    STATUS_OK,
    STATUS_MULTIPLE_CHOICES,
    STATUS_UNPROCESSABLE_ENTITY,
} from '@smoozzy/http-statuses';

export default new Api({
    validateStatus(status) {
        return status >= STATUS_OK && status < STATUS_MULTIPLE_CHOICES ||
            status === STATUS_UNPROCESSABLE_ENTITY;
    },
});
```


## Status codes

### 1xx Informational

 Code | Status Text                        | Description
------|------------------------------------|---------------------------------------------
 100  | Continue                           | [Description](https://httpstatuses.com/100)
 101  | Switching Protocols                | [Description](https://httpstatuses.com/101)
 102  | Processing                         | [Description](https://httpstatuses.com/102)
 103  | Early Hints                        | [Description](https://tools.ietf.org/html/rfc8297)


### 2xx Success

 Code | Status Text                        | Description 
------|------------------------------------|---------------------------------------------
 200  | OK                                 | [Description](https://httpstatuses.com/200)
 201  | Created                            | [Description](https://httpstatuses.com/201)
 202  | Accepted                           | [Description](https://httpstatuses.com/202)
 203  | Non-authoritative Information      | [Description](https://httpstatuses.com/203)
 204  | No Content                         | [Description](https://httpstatuses.com/204)
 205  | Reset Content                      | [Description](https://httpstatuses.com/205)
 206  | Partial Content                    | [Description](https://httpstatuses.com/206)
 207  | Multi-Status                       | [Description](https://httpstatuses.com/207)
 208  | Already Reported                   | [Description](https://httpstatuses.com/208)
 226  | IM Used                            | [Description](https://httpstatuses.com/226)


### 3xx Redirection

 Code | Status Text                        | Description 
------|------------------------------------|---------------------------------------------
 300  | Multiple Choices                   | [Description](https://httpstatuses.com/300)
 301  | Moved Permanently                  | [Description](https://httpstatuses.com/301)
 302  | Found                              | [Description](https://httpstatuses.com/302)
 303  | See Other                          | [Description](https://httpstatuses.com/303)
 304  | Not Modified                       | [Description](https://httpstatuses.com/304)
 305  | Use Proxy                          | [Description](https://httpstatuses.com/305)
 307  | Temporary Redirect                 | [Description](https://httpstatuses.com/307)
 308  | Permanent Redirect                 | [Description](https://httpstatuses.com/308)


### 4xx Client Error

 Code | Status Text                        | Description 
------|------------------------------------|---------------------------------------------
 400  | Bad Request                        | [Description](https://httpstatuses.com/400)
 401  | Unauthorized                       | [Description](https://httpstatuses.com/401)
 402  | Payment Required                   | [Description](https://httpstatuses.com/402)
 403  | Forbidden                          | [Description](https://httpstatuses.com/403)
 404  | Not Found                          | [Description](https://httpstatuses.com/404)
 405  | Method Not Allowed                 | [Description](https://httpstatuses.com/405)
 406  | Not Acceptable                     | [Description](https://httpstatuses.com/406)
 407  | Proxy Authentication Required      | [Description](https://httpstatuses.com/407)
 408  | Request Timeout                    | [Description](https://httpstatuses.com/408)
 409  | Conflict                           | [Description](https://httpstatuses.com/409)
 410  | Gone                               | [Description](https://httpstatuses.com/410)
 411  | Length Required                    | [Description](https://httpstatuses.com/411)
 412  | Precondition Failed                | [Description](https://httpstatuses.com/412)
 413  | Payload Too Large                  | [Description](https://httpstatuses.com/413)
 414  | Request-URI Too Long               | [Description](https://httpstatuses.com/414)
 415  | Unsupported Media Type             | [Description](https://httpstatuses.com/415)
 416  | Requested Range Not Satisfiable    | [Description](https://httpstatuses.com/416)
 417  | Expectation Failed                 | [Description](https://httpstatuses.com/417)
 418  | I'm a teapot                       | [Description](https://httpstatuses.com/418)
 421  | Misdirected Request                | [Description](https://httpstatuses.com/421)
 422  | Unprocessable Entity               | [Description](https://httpstatuses.com/422)
 423  | Locked                             | [Description](https://httpstatuses.com/423)
 424  | Failed Dependency                  | [Description](https://httpstatuses.com/424)
 425  | Too Early                          | [Description](https://tools.ietf.org/html/rfc8470#section-5.2)
 426  | Upgrade Required                   | [Description](https://httpstatuses.com/426)
 428  | Precondition Required              | [Description](https://httpstatuses.com/428)
 429  | Too Many Requests                  | [Description](https://httpstatuses.com/429)
 431  | Request Header Fields Too Large    | [Description](https://httpstatuses.com/431)
 444  | Connection Closed Without Response | [Description](https://httpstatuses.com/444)
 451  | Unavailable For Legal Reasons      | [Description](https://httpstatuses.com/451)
 499  | Client Closed Request              | [Description](https://httpstatuses.com/499)


### 5xx Server Error

 Code | Status Text                        | Description
------|------------------------------------|---------------------------------------------
 500  | Internal Server Error              | [Description](https://httpstatuses.com/500)
 501  | Not Implemented                    | [Description](https://httpstatuses.com/501)
 502  | Bad Gateway                        | [Description](https://httpstatuses.com/502)
 503  | Service Unavailable                | [Description](https://httpstatuses.com/503)
 504  | Gateway Timeout                    | [Description](https://httpstatuses.com/504)
 505  | HTTP Version Not Supported         | [Description](https://httpstatuses.com/505)
 506  | Variant Also Negotiates            | [Description](https://httpstatuses.com/506)
 507  | Insufficient Storage               | [Description](https://httpstatuses.com/507)
 508  | Loop Detected                      | [Description](https://httpstatuses.com/508)
 510  | Not Extended                       | [Description](https://httpstatuses.com/510)
 511  | Network Authentication Required    | [Description](https://httpstatuses.com/511)
 599  | Network Connect Timeout Error      | [Description](https://httpstatuses.com/599)


## Links

- [IANA Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml)
- [HTTP Status Codes](https://httpstatuses.com)


CHANGELOG.md
## [0.1.1](https://github.com/smoozzy/http-statuses/compare/v0.1.0...v0.1.1) (2019-09-08)

### Bugfixes

* Updated documentation for "Too Early" status (425)


## 0.1.0 (2019-09-02)

Initial release


rollup.config.js
// @see https://rollupjs.org/guide/en/

import babel from 'rollup-plugin-babel';
import commonjs from 'rollup-plugin-commonjs';
import replace from 'rollup-plugin-replace';
import resolve from 'rollup-plugin-node-resolve';
import {terser} from 'rollup-plugin-terser';
import pkg from './package.json';


const banner = [
    `/*! Smoozzy HTTP Statuses v${pkg.version}`,
    ` *  ${pkg.homepage}`,
    ' *  Released under the MIT license',
    ' */'
].join('\n');


export default {
    input: 'src/index.js',

    output: [
        {
            format: 'cjs',  // node.js
            file: pkg.main,
            banner,
        },
        {
            format: 'umd',  // browser
            file: pkg.browser,
            name: 'smoozzy/http-statuses',
            banner,
        },
        {
            format: 'iife',  // unpkg
            file: pkg.unpkg,
            name: 'smoozzy.httpstatuses',
            banner,
        }
    ].reduce((memo, config) => {  // add config for minified versions
        memo.push(config);
        memo.push(Object.assign({}, config, {
            file: config.file.replace(/\.js$/, '.min.js'),
        }));

        return memo;
    }, []),

    plugins: [
        resolve(),
        commonjs(),
        replace({
            'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV || 'development'),
        }),
        babel({
            exclude: 'node_modules/**',
        }),
        terser({
            include: [/\.min\.js$/],
            output: {
                comments: /^!/,  // keep banner
            },
        }),
    ],
};


src/index.js
/* 1xx Informational
 */

export const STATUS_CONTINUE = 100;                            // https://tools.ietf.org/html/rfc7231#section-6.2.1
export const STATUS_SWITCHING_PROTOCOLS = 101;                 // https://tools.ietf.org/html/rfc7231#section-6.2.2
export const STATUS_PROCESSING = 102;                          // https://tools.ietf.org/html/rfc2518#section-10.1
export const STATUS_EARLY_HINTS = 103;                         // https://tools.ietf.org/html/rfc8297#section-2


/* 2xx Success
 */

export const STATUS_OK = 200;                                  // https://tools.ietf.org/html/rfc7231#section-6.3.1
export const STATUS_CREATED = 201;                             // https://tools.ietf.org/html/rfc7231#section-6.3.2
export const STATUS_ACCEPTED = 202;                            // https://tools.ietf.org/html/rfc7231#section-6.3.3
export const STATUS_NON_AUTHORITATIVE_INFORMATION = 203;       // https://tools.ietf.org/html/rfc7231#section-6.3.4
export const STATUS_NO_CONTENT = 204;                          // https://tools.ietf.org/html/rfc7231#section-6.3.5
export const STATUS_RESET_CONTENT = 205;                       // https://tools.ietf.org/html/rfc7231#section-6.3.6
export const STATUS_PARTIAL_CONTENT = 206;                     // https://tools.ietf.org/html/rfc7233#section-4.1
export const STATUS_MULTI_STATUS = 207;                        // https://tools.ietf.org/html/rfc4918#section-13
export const STATUS_ALREADY_REPORTED = 208;                    // https://tools.ietf.org/html/rfc5842#section-7.1
export const STATUS_IM_USED = 226;                             // https://tools.ietf.org/html/rfc3229#section-10.4.1


/* 3xx Redirection
 */

export const STATUS_MULTIPLE_CHOICES = 300;                    // https://tools.ietf.org/html/rfc7231#section-6.4.1
export const STATUS_MOVED_PERMANENTLY = 301;                   // https://tools.ietf.org/html/rfc7231#section-6.4.2
export const STATUS_FOUND = 302;                               // https://tools.ietf.org/html/rfc7231#section-6.4.3
export const STATUS_SEE_OTHER = 303;                           // https://tools.ietf.org/html/rfc7231#section-6.4.4
export const STATUS_NOT_MODIFIED = 304;                        // https://tools.ietf.org/html/rfc7232#section-4.1
export const STATUS_USE_PROXY = 305;                           // https://tools.ietf.org/html/rfc7231#section-6.4.5
// export const STATUS_UNUSED = 306;                           // https://tools.ietf.org/html/rfc7231#section-6.4.6
export const STATUS_TEMPORARY_REDIRECT = 307;                  // https://tools.ietf.org/html/rfc7231#section-6.4.7
export const STATUS_PERMANENT_REDIRECT = 308;                  // https://tools.ietf.org/html/rfc7538#section-3


/* 4xx Client Error
 */

export const STATUS_BAD_REQUEST = 400;                         // https://tools.ietf.org/html/rfc7231#section-6.5.1
export const STATUS_UNAUTHORIZED = 401;                        // https://tools.ietf.org/html/rfc7235#section-3.1
export const STATUS_PAYMENT_REQUIRED = 402;                    // https://tools.ietf.org/html/rfc7231#section-6.5.2
export const STATUS_FORBIDDEN = 403;                           // https://tools.ietf.org/html/rfc7231#section-6.5.3
export const STATUS_NOT_FOUND = 404;                           // https://tools.ietf.org/html/rfc7231#section-6.5.4
export const STATUS_METHOD_NOT_ALLOWED = 405;                  // https://tools.ietf.org/html/rfc7231#section-6.5.5
export const STATUS_NOT_ACCEPTABLE = 406;                      // https://tools.ietf.org/html/rfc7231#section-6.5.6
export const STATUS_PROXY_AUTHENTICATION_REQUIRED = 407;       // https://tools.ietf.org/html/rfc7235#section-3.2
export const STATUS_REQUEST_TIMEOUT = 408;                     // https://tools.ietf.org/html/rfc7231#section-6.5.7
export const STATUS_CONFLICT = 409;                            // https://tools.ietf.org/html/rfc7231#section-6.5.8
export const STATUS_GONE = 410;                                // https://tools.ietf.org/html/rfc7231#section-6.5.9
export const STATUS_LENGTH_REQUIRED = 411;                     // https://tools.ietf.org/html/rfc7231#section-6.5.10
export const STATUS_PRECONDITION_FAILED = 412;                 // https://tools.ietf.org/html/rfc7232#section-4.2
export const STATUS_PAYLOAD_TOO_LARGE = 413;                   // https://tools.ietf.org/html/rfc7231#section-6.5.11
export const STATUS_REQUEST_URI_TOO_LONG = 414;                // https://tools.ietf.org/html/rfc7231#section-6.5.12
export const STATUS_UNSUPPORTED_MEDIA_TYPE = 415;              // https://tools.ietf.org/html/rfc7231#section-6.5.13
export const STATUS_REQUESTED_RANGE_NOT_SATISFIABLE = 416;     // https://tools.ietf.org/html/rfc7233#section-4.4
export const STATUS_EXPECTATION_FAILED = 417;                  // https://tools.ietf.org/html/rfc7231#section-6.5.14
export const STATUS_I_AM_A_TEAPOT = 418;                       // https://tools.ietf.org/html/rfc2324#section-2.3.2
export const STATUS_MISDIRECTED_REQUEST = 421;                 // https://tools.ietf.org/html/rfc7540#section-9.1.2
export const STATUS_UNPROCESSABLE_ENTITY = 422;                // https://tools.ietf.org/html/rfc4918#section-11.2
export const STATUS_LOCKED = 423;                              // https://tools.ietf.org/html/rfc4918#section-11.3
export const STATUS_FAILED_DEPENDENCY = 424;                   // https://tools.ietf.org/html/rfc4918#section-11.4
export const STATUS_TOO_EARLY = 425;                           // https://tools.ietf.org/html/rfc8470#section-5.2
export const STATUS_UPGRADE_REQUIRED = 426;                    // https://tools.ietf.org/html/rfc7231#section-6.5.15
export const STATUS_PRECONDITION_REQUIRED = 428;               // https://tools.ietf.org/html/rfc6585#section-3
export const STATUS_TOO_MANY_REQUESTS = 429;                   // https://tools.ietf.org/html/rfc6585#section-4
export const STATUS_REQUEST_HEADER_FIELDS_TOO_LARGE = 431;     // https://tools.ietf.org/html/rfc6585#section-5
export const STATUS_CONNECTION_CLOSED_WITHOUT_RESPONSE = 444;  // https://nginx.org/en/docs/http/ngx_http_rewrite_module.html#return
export const STATUS_UNAVAILABLE_FOR_LEGAL_REASONS = 451;       // https://tools.ietf.org/html/rfc7725#section-3
export const STATUS_CLIENT_CLOSED_REQUEST = 499;               // http://lxr.nginx.org/source/src/http/ngx_http_request.h#0127


/* 5xx Server Error
 */

export const STATUS_INTERNAL_SERVER_ERROR = 500;               // https://tools.ietf.org/html/rfc7231#section-6.6.1
export const STATUS_NOT_IMPLEMENTED = 501;                     // https://tools.ietf.org/html/rfc7231#section-6.6.2
export const STATUS_BAD_GATEWAY = 502;                         // https://tools.ietf.org/html/rfc7231#section-6.6.3
export const STATUS_SERVICE_UNAVAILABLE = 503;                 // https://tools.ietf.org/html/rfc7231#section-6.6.4
export const STATUS_GATEWAY_TIMEOUT = 504;                     // https://tools.ietf.org/html/rfc7231#section-6.6.5
export const STATUS_HTTP_VERSION_NOT_SUPPORTED = 505;          // https://tools.ietf.org/html/rfc7231#section-6.6.6
export const STATUS_VARIANT_ALSO_NEGOTIATES = 506;             // https://tools.ietf.org/html/rfc2295#section-8.1
export const STATUS_INSUFFICIENT_STORAGE = 507;                // https://tools.ietf.org/html/rfc4918#section-11.5
export const STATUS_LOOP_DETECTED = 508;                       // https://tools.ietf.org/html/rfc5842#section-7.2
export const STATUS_NOT_EXTENDED = 510;                        // https://tools.ietf.org/html/rfc2774#section-7
export const STATUS_NETWORK_AUTHENTICATION_REQUIRED = 511;     // https://tools.ietf.org/html/rfc6585#section-6
export const STATUS_NETWORK_CONNECT_TIMEOUT_ERROR = 599;       // https://github.com/rmaake1/httpstatuses/issues/22
